lib/zhr_devs/submissions/task.ex

```diff
@@ -8,41 +8,100 @@ defmodule ZhrDevs.Submissions.Task do
 
   @dialyzer {:nowarn_function, [new: 0, new: 1, new: 2, new: 3]}
 
+  @supported_technologies :zhr_devs
+                          |> Application.compile_env(:supported_technologies)
+                          |> Enum.map(&Atom.to_string/1)
+
   import Algae
 
   @derive Jason.Encoder
 
+  defmodule Language do
+    @moduledoc "The supported language in which task is written"
+    @derive Jason.Encoder
+    defdata(Uptight.Text.t())
+
+    alias Uptight.Text, as: T
+
+    def new!(binary), do: %Language{language: T.new!(binary)}
+
+    def from_raw!(%Language{language: language}) do
+      T.un(language)
+    end
+  end
+
+  defmodule Library do
+    @moduledoc "The libraries used in task"
+    @derive Jason.Encoder
+    defdata(Uptight.Text.t())
+
+    alias Uptight.Text, as: T
+
+    def new!(binary), do: %Library{library: T.new!(binary)}
+
+    def from_raw!(%Library{library: library}) do
+      T.un(library)
+    end
+  end
+
+  defmodule Integration do
+    @moduledoc false
+
+    alias Uptight.Text, as: T
+
+    @derive Jason.Encoder
+    defdata(Uptight.Text.t())
+
+    def new!(binary), do: %Integration{integration: T.new!(binary)}
+
+    def from_raw!(%Integration{integration: integration}) do
+      T.un(integration)
+    end
+  end
+
+  defimpl Jason.Encoder, for: [Language, Library, Integration] do
+    def encode(%struct{} = value, opts) do
+      value |> struct.from_raw!() |> Jason.Encode.string(opts)
+    end
+  end
+
   defdata do
     task_name :: Uptight.Text.t()
-    programming_language :: Uptight.Text.t()
-    integrations :: list(Uptight.Text.t()) \\ []
-    library_stack :: list(Uptight.Text.t()) \\ []
+    programming_language :: Language.t()
+    integrations :: list(Integration.t()) \\ []
+    library_stack :: list(Library.t()) \\ []
   end
 
-  @required_keys [:task_name, :programming_language]
+  @list_keys [:integrations, :library_stack]
 
-  @spec from_uri(binary()) :: __MODULE__.t()
-  def from_uri(task_from_uri) when is_binary(task_from_uri) do
+  @spec new!(binary()) :: __MODULE__.t()
+  def new!(task_from_uri) do
```

Comment:
This *really* does not deserve to be a default constructor. I expect default constructor to be `def new!(name, language, libraries, integrations)`. In that order, because they go from more specific to less specific things.

I would also rename `library_stack` into `libraries` unless atom `:libraries` has a special meaning in Elixir.